tests/cupy_tests/logic_tests/test_comparison.py

```diff
@@ -97,6 +97,25 @@ def test_array_equal_diff_length(self, xp, dtype):
         b = xp.array([1, 2, 3], dtype=dtype)
         return xp.array_equal(a, b)
 
+    @testing.with_requires('numpy>=1.19')
+    @testing.for_float_dtypes()
+    @testing.numpy_cupy_equal()
+    def test_array_equal_infinite_equal_nan(self, xp, dtype):
+        nan = float('nan')
+        inf = float('inf')
+        ninf = float('-inf')
+        a = xp.array([0, nan, inf, ninf], dtype=dtype)
+        b = xp.array([0, nan, inf, ninf], dtype=dtype)
+        return xp.array_equal(a, b, equal_nan=True)
+
+    @testing.with_requires('numpy>=1.19')
+    @testing.for_complex_dtypes()
+    @testing.numpy_cupy_equal()
+    def test_array_equal_complex_equal_nan(self, xp, dtype):
+        a = xp.array([1+2j, 3+4j], dtype=dtype)
+        b = xp.array([1+2j, 3+4j], dtype=dtype)
```

Comment:
from numpy docs 
```
Whether to compare NaN’s as equal. If the dtype of a1 and a2 is complex, values will be considered equal if either the real or the imaginary component of a given value is nan.

New in version 1.19.0.
```

Please test complex numbers with `nan` in their real or imaginary parts 🙂

Reply:
I missed this, thanks! 👍🏼
I updated the PR.